domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ActivityLifecycleObserver.kt

```diff
@@ -53,7 +52,7 @@ class ActivityLifecycleObserver @Inject constructor(
 
   override fun onActivityResumed(activity: Activity) {
     val currentAppScreen = activity.intent.getProtoExtra(
```

Comment:
I think you could have a simple getter on the decorator which injects the current activity, retrieves the intent, and then retrieves the screen name from that intent. That would simplify retrieval.

Reply:
I've introduced an extension function which will act as a getter. For consistency between screen name packing and retrieval, I've opted to use the intent instead of injecting the activity. Does that make sense?

Reply:
Yep, the new approach looks good. Thanks!